draft-peabody-dispatch-new-uuid-format-03.txt

```diff
@@ -321,1105 +322,694 @@ Internet-Draft               new-uuid-format               February 2022
+TODO: some C code here showing how to generate, including option for counter
 
-   1.  The unix timestamp defined as unixts
-
-   2.  The sub-second precision values defined as subsec_a, subsec_b,
-       and subsec_seq_node
-
-   As detailed in Figure 2 the unix timestamp (unixts) is always the
-   first 36 bits of the UUIDv7 layout.
-
-
-
-
-Peabody & Davis           Expires 5 August 2022                [Page 15]
-
-Internet-Draft               new-uuid-format               February 2022
+                    Figure 5: UUIDv7 Function in C
 
+5.3.  Example of a UUIDv7 Value
 
-   Similarly as per Figure 2, the sub-second precision values lie within
-   subsec_a, subsec_b, and subsec_seq_node which are all interpreted as
-   sub-second information after skipping over the version (ver) and
-   (var) bits.  These concatenated sub-second information bits are
-   interpreted in a way where most to least significant bits represent a
-   further division by two.  This is the same normal place notation used
-   to express fractional numbers, except in binary.  For example, in
-   decimal ".1" means one tenth, and ".01" means one hundredth.  In this
-   subsec field, a 1 means one half, 01 means one quarter, 001 is one
-   eighth, etc.  This scheme can work for any number of bits up to the
-   maximum available, and keeps the most significant data leftmost in
-   the bit sequence.
+   TODO: Show a few UUIDv7 values with the field/value breakdown
 
-   To perform the sub-second math, simply take the first (most
-   significant/leftmost) N bits of subsec and divide it by 2^N.  Take
-   for example:
+   -------------------------------
+   field      bits    value
+   -------------------------------
+   unix_ts_ms   48    <Example>
+   rand_a       16    <Example>
+   var_ver       8    0xE7
+   rand_b       56    <Example>
+   -------------------------------
+   total        128
+   -------------------------------
+   final: <example_uuidv7_from_table_inputs>
 
-   1.  To parse the first 16 bits, extract that value as an integer and
-       divide it by 65536 (2 to the 16th).
+                    Figure 6: UUIDv7 Example Test Vector
 
-   2.  If these 16 bits are 0101 0101 0101 0101, then treating that as
-       an integer gives 0x5555 or 21845 in decimal, and dividing by
-       65536 gives 0.3333282
+6.  UUID Version 8
 
-   This sub-second encoding scheme provides maximum interoperability
-   across systems where different levels of time precision are
-   required/feasible/available.  The timestamp value derived from a
-   UUIDv7 value SHOULD be "as close to the correct value as possible"
-   when parsed, even across disparate systems.
+6.1.  Format
 
-   Take for example the starting point for our next two UUIDv7 parsing
-   scenarios:
+   UUID version 8 provides a backward compatible UUID format which
+   otherwise has no restrictions on format except that the variant bits
+   and version bits be set to 0xE8 in octet 9 as defined in Section 7.2.
```

Comment:
Be explicit about the fact this is intended for  experimental and/or vendor-specific formats?
```suggestion
   UUID version 8 provides an RFC-compatible format for experimental or vendor-specific use cases.  The only requirement is that the variant and version bits be set as defined in Section 7.2 (version set to 0100b).
   
  Version 8 UUID uniqueness will be implementation-specific.
```